catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/SecureHadoopCatalogOperations.java

```diff
@@ -0,0 +1,312 @@
+    UserGroupInformation currentUser = getUGIByIdent(properties, ident);
+    return doAs(
+        currentUser,
+        () ->
+            hadoopCatalogOperations.createFileset(
+                ident, comment, type, storageLocation, properties),
+        ident);
+  }
+
+  @Override
+  public boolean dropFileset(NameIdentifier ident) {
+    FilesetEntity filesetEntity;
+    try {
+      filesetEntity =
+          hadoopCatalogOperations
+              .getStore()
+              .get(ident, Entity.EntityType.FILESET, FilesetEntity.class);
+    } catch (NoSuchEntityException e) {
+      LOG.warn("Fileset {} does not exist", ident);
+      return false;
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to delete fileset " + ident, ioe);
+    }
+
+    // Reset the current user based on the name identifier.
+    UserGroupInformation currentUser = getUGIByIdent(filesetEntity.properties(), ident);
+
+    boolean r = doAs(currentUser, () -> hadoopCatalogOperations.dropFileset(ident), ident);
+    cleanUserInfo(ident);
+    return r;
+  }
+
+  @Override
+  public Schema createSchema(NameIdentifier ident, String comment, Map<String, String> properties)
+      throws NoSuchCatalogException, SchemaAlreadyExistsException {
+    // Reset the current user based on the name identifier and properties.
+    UserGroupInformation currentUser = getUGIByIdent(properties, ident);
+
+    return doAs(
+        currentUser, () -> hadoopCatalogOperations.createSchema(ident, comment, properties), ident);
+  }
+
+  @Override
+  public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
+    try {
+      SchemaEntity schemaEntity =
+          hadoopCatalogOperations
+              .getStore()
+              .get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+      Map<String, String> properties =
+          Optional.ofNullable(schemaEntity.properties()).orElse(Collections.emptyMap());
+
+      // Reset the current user based on the name identifier.
+      UserGroupInformation user = getUGIByIdent(properties, ident);
+
+      boolean r = doAs(user, () -> hadoopCatalogOperations.dropSchema(ident, cascade), ident);
+      cleanUserInfo(ident);
+      return r;
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to delete schema " + ident, ioe);
+    }
+  }
+
+  @Override
+  public void initialize(
+      Map<String, String> config, CatalogInfo info, HasPropertyMetadata propertiesMetadata)
+      throws RuntimeException {
+    hadoopCatalogOperations.initialize(config, info, propertiesMetadata);
+  }
+
+  @Override
+  public Fileset alterFileset(NameIdentifier ident, FilesetChange... changes)
+      throws NoSuchFilesetException, IllegalArgumentException {
+    try {
+      return hadoopCatalogOperations.alterFileset(ident, changes);
+    } finally {
+      String finalName = ident.name();
+      for (FilesetChange change : changes) {
```

Comment:
There two rename operations.
```
Rename A succesffully,
Rename B failed.
```

Reply:
This scenario can not occur because we will merge the multiple rename operations into one and the last rename request will take effect. 

https://github.com/apache/gravitino/blob/b2a93022553a258a8f599f2e5a6e0bbdc3e5ace7/catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java#L589-L606

Reply:
If rename operation failed, is it right?